state/application.go

```diff
@@ -1005,6 +1005,60 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string) (err error) {
+	// Make sure no units are added or removed while the update-series
+	// transaction is being executed. To cover the case where a new subordinate
+	// may be added.
+	units, err := a.AllUnits()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var unitOps []txn.Op
+	for _, u := range units {
+		subNames := u.SubordinateNames()
+		unitOps = append(unitOps, txn.Op{
+			C:      unitsC,
+			Id:     u.doc.DocID,
+			Assert: bson.D{{"subordinates", subNames}},
+		})
+		unitOps = append(unitOps, txn.Op{
+			C:      unitsC,
+			Id:     u.doc.DocID,
+			Assert: txn.DocExists,
+		})
+	}
+	unitOps = append(unitOps, txn.Op{
+		C:      applicationsC,
+		Id:     a.doc.DocID,
+		Assert: bson.D{{"unitcount", len(units)}},
+	})
+
+	// Make sure that the charm doesn't change while updating the series
+	// the application.
+	chURL := a.doc.CharmURL
+	chURLOp := txn.Op{
+		C:      applicationsC,
+		Id:     a.doc.DocID,
+		Assert: bson.D{{"charmurl", chURL}},
+	}
+
+	var ops []txn.Op
```

Comment:
The ops here should also include the ones to update any subordinates, instead of doing that in the apiserver layer.